core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -961,7 +961,7 @@ private[spark] class TaskSetManager(
       taskSetExcludelistHelperOpt.foreach(_.updateExcludedForFailedTask(
         info.host, info.executorId, index, failureReason))
       numFailures(index) += 1
```

Comment:
IIUC we should not increase the `numFailures` if the task is already succeeded.

Reply:
> IIUC we should not increase the `numFailures` if the task is already succeeded.

Yeah, thanks for your advice, I`ll handle this.

Reply:
> IIUC we should not increase the `numFailures` if the task is already succeeded.

PTAL when time free, thanks